.github/workflows/push.yml

```diff
@@ -0,0 +1,23 @@
+name: buf-push
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: bufbuild/buf-setup-action@v1
+      - uses: bufbuild/buf-lint-action@v1
+        with:
+          input: "api"
+      - uses: bufbuild/buf-breaking-action@v1
```

Comment:
Just thinking it through, what is supposed to happen here if buf breaking detects breaking changes? Presumably if this is the point of PR merge then we are saying that we are accepting the breaking changes and we want to merge them?

Reply:
Breaking is more of like a warning to let us know of possibly breaking clients. So the examples would be changing a field's name or entirely removing a field, it's not meant to block a merge but let you know if there could be a problem so you can resolve the issue.

Reply:
Yeah, and I can definitely see the benefit of having it in the PR workflow, since this is where we want it flagged, but if the push workflow happens on PR merge, then I'm not sure what value there is to looking for breaking changes at this point. I mean, from time to time, we will make breaking changes on purpose. I'm not clear about how it would actually inform the user at this point.

Still, if this is what the docs recommend, we can just push forward and see how it works in practice. Easy to change later. :)